cpp/oneapi/dal/algo/basic_statistics/backend/cpu/compute_kernel.cpp

```diff
@@ -187,21 +187,30 @@ result_t call_daal_kernel_without_weights(const context_cpu& ctx,
     daal_input.set(daal_lom::InputId::data, daal_data);
 
     interop::status_to_exception(
-        daal_result.allocate<Float>(&daal_input, &daal_parameter, get_daal_method<method_t>()));
-
-    interop::status_to_exception(
-        interop::call_daal_kernel<Float, daal_lom_batch_kernel_t>(ctx,
-                                                                  daal_data.get(),
-                                                                  &daal_result,
-                                                                  &daal_parameter));
+        daal_result.allocate<Float>(&daal_input, &daal_parameter, daal_method));
+
+    if (std::is_same_v<method::dense, Method>) {
+        interop::status_to_exception(
+            interop::call_daal_kernel<Float, dense_batch_kernel_t>(ctx,
+                                                                   daal_data.get(),
+                                                                   &daal_result,
+                                                                   &daal_parameter));
+    }
+    else {
+        interop::status_to_exception(
+            interop::call_daal_kernel<Float, sparse_batch_kernel_t>(ctx,
+                                                                    daal_data.get(),
+                                                                    &daal_result,
+                                                                    &daal_parameter));
+    }
```

Comment:
This if can be omitted in case `batch_kernel_t` is used for both dense and sparse kernels.